firmware/usb.c

```diff
@@ -426,9 +427,10 @@ char usbTiny(char set)
 	return old;
 }
 
-void usbDelay(int cycles)
+void usbSleep(uint32_t millis)
 {
-	while (cycles--) {
+	uint32_t end = system_millis + millis;
+	while (end > system_millis) {
```

Comment:
I did calculate this but considered it non-critical, as you did, since churning out more OpenPGP code is currently my priority.

But a solution (like the ones you posted) would be nice :wink: